backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/Migrations.java

```diff
@@ -0,0 +1,51 @@
+package org.molgenis.emx2.sql;
+
+import java.io.IOException;
+import org.molgenis.emx2.Database;
+import org.molgenis.emx2.MolgenisException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Migrations {
+  // version the current software needs to work
+  private static final int SOFTWARE_DATABASE_VERSION = 1;
+  private static Logger logger = LoggerFactory.getLogger(Migrations.class);
+
+  public static synchronized void initOrMigrate(SqlDatabase db) {
+    db.tx(
```

Comment:
transaction creates locks , not sure this if still feasible , with non trivial data set and lost of migrations, but i guess will can use this until we get a max locks error.

Reply:
indeed.